src/main/java/com/hotels/hcommon/hive/metastore/core/iterator/BatchResolver.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2017-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.hcommon.hive.metastore.core.iterator;
```

Comment:
I think keeping "iterator" as a sub-package is fine, just without the core. I do wonder then if it's worth grouping some of the other classes in sub-packages instead of keeping them all at the root? Thoughts?